gdx/src/com/badlogic/gdx/utils/DefaultPool.java

```diff
@@ -0,0 +1,30 @@
+
+package com.badlogic.gdx.utils;
+
+/** Default Pool implementation that creates a new instances of a type based on a supplier. */
+public class DefaultPool<T> extends Pool<T> {
+
+	private final PoolSupplier<T> poolTypeSupplier;
+
+	public DefaultPool (PoolSupplier<T> supplier) {
+		this(supplier, 16, Integer.MAX_VALUE);
+	}
+
+	public DefaultPool (PoolSupplier<T> supplier, int initialCapacity) {
+		this(supplier, initialCapacity, Integer.MAX_VALUE);
+	}
+
+	public DefaultPool (PoolSupplier<T> supplier, int initialCapacity, int max) {
+		super(initialCapacity, max);
+		this.poolTypeSupplier = supplier;
+	}
+
+	@Override
+	protected T newObject () {
+		return poolTypeSupplier.get();
+	}
+
+	public interface PoolSupplier<T> {
```

Comment:
Just a little documentation would help here. This is the first recommended usage of Java 8's constructor reference syntax in all of libGDX, so showing how to use this might help.

Reply:
I can add it, but it's really just standard java syntax at this point.

Reply:
It is for normal Java development, but libGDX is still targeting Java 7 at this point in time.